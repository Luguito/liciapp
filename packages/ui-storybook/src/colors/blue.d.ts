import { CommonColor } from './commonColor.d'

export interface BlueColor extends CommonColor {
    1: string;
    2: string;
    3: string;
    4: string;
    5: string;
    dark: string;
    light: string;
    navy: string;
    threeTerms: string;
    sixTerms: string;
    greaterTerms: string;
    stepperLightBlue?: string;
}
