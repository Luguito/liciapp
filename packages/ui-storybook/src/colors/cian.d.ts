import { CommonColor } from './commonColor.d'

export interface CianColor extends CommonColor {
    dark: string;
}
