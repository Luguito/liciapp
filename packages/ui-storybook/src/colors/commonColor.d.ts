export interface CommonColor {
    normal: string;
}
