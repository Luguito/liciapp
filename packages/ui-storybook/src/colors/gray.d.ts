import { CommonColor } from './commonColor.d'

export interface GrayColor extends CommonColor {
    obscure: string;
    dark: string;
    light: string;
    highLight: string;
    background: string;
    backgroundDark: string;
}
