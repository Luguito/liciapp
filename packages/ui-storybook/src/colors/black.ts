import { BlackColor } from './black.d';

const Black: BlackColor = {
    normal: '#000000'
};

export default Black;
