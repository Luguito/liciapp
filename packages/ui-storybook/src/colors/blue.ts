import { BlueColor } from './blue.d';

const Blue: BlueColor  = {
    1: '#3154c5',
    2: '#3c65ec',
    3: '#6384eb',
    4: '#b1c1f5',
    5: '#f5f9fc',
    dark: '#202844',
    navy: '#f8fafb',
    light: '#3154c5',
    normal: '#334375',
    sixTerms: '#5886FF',
    threeTerms: '#4069DB',
    greaterTerms: '#2D52B4',
    stepperLightBlue: '#ecf0fc'
};

export default Blue;
