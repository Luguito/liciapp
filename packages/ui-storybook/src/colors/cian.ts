import { CianColor } from './cian.d';

const Cian: CianColor = {
    normal: '#00c3d3',
    dark: '#00b2c1'
};

export default Cian;
