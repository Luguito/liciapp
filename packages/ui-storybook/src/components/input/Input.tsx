import React from 'react';
import { InputProps, TextAreaProps } from './input.d';
import { TextInput } from './TextInput';
import { TextAreaInput } from './TextAreaInput';

export { TEXT_ALIGN } from './TextInput.styled';

export default (props: InputProps | TextAreaProps) => {
    return (
        <>
            {props.multiline ? <TextAreaInput {...props} /> : <TextInput {...props} />}
        </>
    )
};
