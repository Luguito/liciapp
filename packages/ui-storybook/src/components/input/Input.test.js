import React from 'react';
import { shallow, mount } from 'enzyme';
import { TextInput } from "./Input";
import { Gray } from "../../colors";

describe("Input", () => {
    describe("Text", () => {
        it ("Snapshot", () => {
            const comp = shallow(<TextInput />);
            expect(comp).toMatchSnapshot()
        })

        it("Should have valid state for default", ()  => {
            const comp = mount(<TextInput placeholder="Say greetings" defaultValue="hello!" />);
            expect(comp).toHaveStyleRule('font-family', 'Barlow');
            expect(comp.find("input")).toHaveStyleRule('border-radius', '3px');
            expect(comp.find("input")).toHaveStyleRule('border', `solid 1px ${Gray.normal}`);
        })

        it("onChanged should be called", () => {
            const onChanged = jest.fn();
            const comp = mount(<TextInput onChange={onChanged} />);

            comp.find('input').simulate("change");
            expect(onChanged).toHaveBeenCalled();
        })

        it("Should have valid attrs", () => {
            const comp = mount(<TextInput defaultValue="hi" maxLength={10} textAlign="right" />);

            expect(comp.find('input').prop("value")).toEqual("hi")
            expect(comp.find('input').prop("maxLength")).toEqual(10)
            expect(comp.find('input')).toHaveStyleRule('text-align', "left")
        })
    });

    describe("Text with Prefix", () => {
        it("Should render the prefix", () => {
            const comp = mount(<TextInput prefix="C.C." />);
            expect(comp.containsMatchingElement(<span>C.C.</span>)).toEqual(true);
        })
    })
});
