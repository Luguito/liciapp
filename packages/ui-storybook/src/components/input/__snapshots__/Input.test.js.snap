// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Text Snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.input
      autoComplete="off"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      textAlign="left"
      type="text"
      value=""
      withPrefix={false}
    />
    <styled.label
      showed={false}
    >
      <styled.span />
    </styled.label>
  </styled.div>
  <styled.div />
</styled.div>
`;
